lib/vu_state.go

```diff
@@ -19,6 +19,11 @@ import (
 // DialContexter is an interface that can dial with a context
 type DialContexter interface {
 	DialContext(ctx context.Context, network, addr string) (net.Conn, error)
+
+	// ResolveAddr looks up the IP address for the given host using the same logic as DialContext.
+	// The address is expected in the form "host:port" or just "host".
+	// It returns the resolved IP, the port (0 if not specified), and an error if any.
+	ResolveAddr(addr string) (net.IP, int, error)
```

Comment:
I do not like the idea that we add more functionality to this interface, that is named specifically after the other funcitonality it has.

To be honest the whole lib.State is not particularly nice way of providing the API as it is just a struct with fields. This works great for internal use, but for public APIs it just lets user do too many "bad" things. 

I am not certain what the best approach will be, but maybe having lib.State have a method that returns this instead of just extending this interface will work better. This likely will still require that the Dialer exposes this functionality, but at least we won't make this interface uglier to implement this.